base/src/main/java/proguard/evaluation/value/InstructionOffsetValue.java

```diff
@@ -60,6 +64,18 @@ public InstructionOffsetValue(int value)
     public InstructionOffsetValue(int[] values)
     {
         this.values = values;
+        // If there are a lot of values, it's faster to use a set
+        // for the "contains" method, than doing a linear search
+        // The threshold for this is chosen by experimentation
+        // to find a balance between memory footprint and performance
+        if (values.length > 100)
+        {
+            valuesMap = new HashSet<>();
```

Comment:
Our CI build issues about OutOfMemory with Proguard brought me here.
https://github.com/line/armeria/pull/4636#issuecomment-1416697145
- To avoid additional hash table expansion, it would be more efficient to specify the initial capacity `new HashSet<>(values.length)`.
- A `int` specific `Set` such as https://fastutil.di.unimi.it/docs/it/unimi/dsi/fastutil/ints/IntOpenHashSet.html may have a small memory footprint without auto boxing.